src/groovy/org/pih/warehouse/importer/CSVUtils.groovy

```diff
@@ -149,4 +150,19 @@ class CSVUtils {
     static CSVPrinter getCSVPrinter() {
         return new CSVPrinter(new StringBuilder(), CSVFormat.DEFAULT)
     }
+
+    /**
+    * Return a CSV separator character based on provided data and expected number of columns.
+    *
+    * Depending on configuration of users preferred csv editor like Excel or Libre office,
+    * when saving a csv file it might reformat the file with a different data separator.
+    * Using this utils function we want to figure out which character is used as a data separator in the file.
+    * */
+    static char getSeparator(String text, Integer columNumber) {
+        String firstLine = text.split("\\r?\\n").first()
+        if (firstLine.contains(Constants.CUSTOM_COLUMN_SEPARATOR) && firstLine.split(Constants.CUSTOM_COLUMN_SEPARATOR).size() == columNumber) {
```

Comment:
Unfortunately my previous version of `getSeparator()` function would fail in this scenario, so after thinking a bit about it I figured out probably a better solution which involves "sample parsing" the first line and checking if the amount of columns match the expected number

https://github.com/openboxes/openboxes/blob/0179626d25ca5806b4124b8ba39086ed1e96e295/src/groovy/org/pih/warehouse/importer/CSVUtils.groovy#L168-L180

The idea behind this solution is to take the first line as a sample for test, usually the first line is a header columns row.
We check if the row contains "the other" separator character, for example a `;`
if it does then we try to parse this line and generate field keys using `initFieldkEys()` function that CSVMapReader plugin provides us with.
If the amount of generated field keys match the expected amount of columns passed as an argument to the function `columnCount` then we can assume that we found the data separator otherwise we fallback on the default `,`